### PR TITLE
add the design of kmesh support workload model

### PR DESCRIPTION
**Pull request by kwb0523**

### CODE REVIEW

File: docs/proposal/service_info_operation.png
Comment (nlgwcy) [inline docs/proposal/service_info_operation.png:1]: The diagram is too complex. If it involves multiple processes, please split it into multiple diagrams.

File: docs/proposal/kmesh_support_workload.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  // Address represents the IPv4/IPv6 address for the workload, this should be globally unique.
+  Addresses [][]byte `protobuf:"bytes,3,rep,name=addresses,proto3" json:"addresses,omitempty"`
+  // The services for which this workload is an endpoint. the key is the NamespacedHostname string of the format namespace/hostname.
+  Services map[string]*PortList `protobuf:"bytes,22,rep,name=services,proto3" json:"services,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
+  // Health status of the workload，such as: "Healthy"，"Unhealthy"
+  Status WorkloadStatus `protobuf:"varint,17,opt,name=status,proto3,enum=istio.workload.WorkloadStatus" json:"status,omitempty"`
+}
+```
+
+## How kmesh subscribe and use the workload model info
```
Comment (nlgwcy) [inline docs/proposal/kmesh_support_workload.md:61]: In the workload model, there are also related processes such as authentication and TLS, which do not seem to be mentioned in the current design. If it is just a basic process design for now, please specify this in the documentation.

File: docs/proposal/kmesh_support_workload.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    __be32 ipv4; // backend ip
+    __u32 port_count;
+    port_pair portlist[MAX_COUNT];
+} __attribute__((packed)) backend_value;
+
+```
+
+## Subscription data process flow
+
+![workload info operation](./workload_info_operation.png)
```
Comment (nlgwcy) [inline docs/proposal/kmesh_support_workload.md:130]: In addition to the design diagrams, please provide a detailed description of the processing logic for each step in the entire workflow, including any potential constraints and limitations.

File: docs/proposal/kmesh_subscribe_workload.png
Comment (hzxuzhonghu) [inline docs/proposal/kmesh_subscribe_workload.png:1]: can you add a svg, so we may edit it in future

File: docs/proposal/kmesh_support_workload.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+## How kmesh subscribe and use the workload model info
+
+1. kmesh's xds client subscribes to the workload model from Istiod through Delta manner, and the type_url is: "type.googleapis.com/istio.workload.Address";
+2. The workload data of Address type is divided into two sub resources: Service and Workload, which are responsed to kmesh and then parsed and converted into internal structures, stored in the bpf map;
+
+![kmesh subscribe workload](./kmesh_subscribe_workload.png)
+
+In the subsequent traffic management of kmesh, based on the IP and Port accessed by the client, find the corresponding service and its endpoints from the BPF map, and then randomly select an endpoint to forward the request to that endpoint;
```
Comment (hzxuzhonghu) [inline docs/proposal/kmesh_support_workload.md:68]: randonly select an endpoint is just current behavior, we should support more LB algorithms

File: docs/proposal/kmesh_support_workload.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+typedef struct
+{
+    __u32 service_id; // service id, through <namespace>/<hostname> string convert to uint32 variable
+} __attribute__((packed)) service_info_value;
+
+// service map
+typedef struct
+{
+    __u32 service_id; // service id, through <namespace>/<hostname> string convert to uint32 variable
```
Comment (hzxuzhonghu) [inline docs/proposal/kmesh_support_workload.md:88]: what's the diff with L82

File: docs/proposal/kmesh_support_workload.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+typedef struct {
+    __be16 service_port;
+    __be16 target_port;
+} __attribute__((packed)) port_pair;
+
+typedef struct
+{
+    __be32 ipv4; // backend ip
+    __u32 port_count;
+    port_pair portlist[MAX_COUNT];
```
Comment (hzxuzhonghu) [inline docs/proposal/kmesh_support_workload.md:123]: Is `MAX_COUNT` a const?

**Comment by hzxuzhonghu** [comment]
@kwb0523 any update

**Pull request merged**